include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,397 @@
+            ((pluginPrefix + ".qz_max").c_str(), po::value<float_64 > (&q_max[2])->default_value(5), "reciprocal space range qz_max (A^-1)")
+            ((pluginPrefix + ".qx_min").c_str(), po::value<float_64 > (&q_min[0])->default_value(-5), "reciprocal space range qx_min (A^-1)")
+            ((pluginPrefix + ".qy_min").c_str(), po::value<float_64 > (&q_min[1])->default_value(-5), "reciprocal space range qy_min (A^-1)")
+            ((pluginPrefix + ".qz_min").c_str(), po::value<float_64 > (&q_min[2])->default_value(-5), "reciprocal space range qz_min (A^-1)")
+            ((pluginPrefix + ".n_qx").c_str(), po::value<unsigned int> (&n_qx)->default_value(100), "Number of qx")
+            ((pluginPrefix + ".n_qy").c_str(), po::value<unsigned int> (&n_qy)->default_value(100), "Number of qy")
+            ((pluginPrefix + ".n_qz").c_str(), po::value<unsigned int> (&n_qz)->default_value(1), "Number of qz");
+    }
+
+
+    std::string pluginGetName() const
+    {
+        return pluginName;
+    }
+
+    void setMappingDescription(MappingDesc *cellDescription)
+    {
+        this->cellDescription = cellDescription;
+    }
+
+    void restart(uint32_t timeStep, const std::string restartDirectory)
+    {
+        // Keep this empty
+    }
+
+    void checkpoint(uint32_t timeStep, const std::string restartDirectory)
+    {
+        // Keep this empty
+    }
+
+private:
+
+    /**
+     * The plugin is loaded on every MPI rank, and therefor this function is
+     * executed on every MPI rank.
+     * One host with MPI rank 0 is defined to be the master.
+     * It creates a folder where all the
+     * results are saved in a plain text format.
+     */
+    void pluginLoad()
+    {
+        if(!notifyPeriod.empty())
+        {
+            /*only rank 0 create a file*/
+            isMaster = reduce.hasResult(mpi::reduceMethods::Reduce());
+            n_q = n_qx * n_qy * n_qz;
+            sumfcoskr = new GridBuffer<float1_64, DIM1 > (DataSpace<DIM1> (n_q));
+            sumfsinkr = new GridBuffer<float1_64, DIM1 > (DataSpace<DIM1> (n_q));
+            np = new GridBuffer<float1_X, DIM1 > (DataSpace<DIM1> (1)); //create one float on GPU and host
+            nmp = new GridBuffer<int64_t, DIM1 > (DataSpace<DIM1> (1)); //create one int on GPU and host
+            sumfcoskr_master = new float1_64[n_q];
+            sumfsinkr_master = new float1_64[n_q];
+            intensity_master = new float1_64[n_q];
+            Environment<>::get().PluginConnector().setNotificationPeriod(this, notifyPeriod);
+            pmacc::Filesystem<simDim>& fs = Environment<simDim>::get().Filesystem();
+
+            if(isMaster)
+            {
+                fs.createDirectory("saxsOutput");
+                fs.setDirectoryPermissions("saxsOutput");
+            }
+        }
+    }
+
+    void writeQintensity(float1_64* intensity, std::string name)
```

Comment:
please add a Doxygen line for the use of the member function and its parameters

Reply:
also `writeQIntensity` for correct camel case

Reply:
> also `writeQIntensity` for correct camel case

Replace it with `writeIntensity` by just deleting the useless 'Q'.